components/slices/MultipleUse.vue

```diff
@@ -0,0 +1,148 @@
+              :field="item.item_title"
+            />
+            <prismic-rich-text
+              class="block-wrapper-item__description"
+              :field="item.item_description"
+            />
+          </div>
+        </div>
+      </div>
+    </div>
+  </section>
+</template>
+
+<script>
+import {
+  EXTRA_LARGE_SCREEN_MIN_WIDTH,
+  DESKTOP_MIN_WIDTH,
+} from '~/config/breakpoints'
+
+export default {
+  name: 'MultipleUse',
+  props: {
+    slice: {
+      type: Object,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      rows: [],
+    }
+  },
+  computed: {
+    background() {
+      return {
+        backgroundColor: `${this.slice.primary.block_background_color}`,
+      }
+    },
+    items() {
+      return this.slice.items
+    },
+    shouldDisplayTitle() {
+      return this.slice.primary.block_should_display_title
+    },
+    shouldUseSubtitle() {
+      return this.slice.primary.block_should_use_subtitle
+    },
+    hasTitle() {
+      return (
+        this.slice.primary.block_title &&
+        this.slice.primary.block_title.length &&
+        this.slice.primary.block_title[0].text &&
+        this.slice.primary.block_title[0].text.length
+      )
+    },
+    hasSubtitle() {
+      return (
+        this.slice.primary.block_subtitle &&
+        this.slice.primary.block_subtitle.length &&
+        this.slice.primary.block_subtitle[0].text &&
+        this.slice.primary.block_subtitle[0].text.length
+      )
+    },
+    title() {
+      return this.slice.primary.block_title
+    },
+    subtitle() {
+      return this.slice.primary.block_subtitle
+    },
+  },
+  mounted() {
+    this.replaceBlockClass()
+    this.rows = this.splitItemsIntoRows()
+  },
+  methods: {
+    hasImage(item) {
+      const isLengthDifferentThanZero = Object.keys(item.item_image).length
+      const isObjectConstructor = item.item_image.constructor === Object
+      return isLengthDifferentThanZero && isObjectConstructor
+    },
+    findNbItemsInRow() {
+      if (process.client) {
+        const isLargeScreen = document.body.clientWidth >= DESKTOP_MIN_WIDTH
+        const isExtraLargeScreen =
+          document.body.clientWidth >= EXTRA_LARGE_SCREEN_MIN_WIDTH
+
+        if (this.items.length === 3 && isLargeScreen) {
+          return 3
+        }
+        if (this.items.length === 4) {
+          if (isExtraLargeScreen) {
+            return 4
+          }
+          if (isLargeScreen) {
+            return 3
+          }
+        }
+        return 2
+      }
+    },
+    splitItemsIntoRows() {
+      const rows = []
+      const chunkSize = this.findNbItemsInRow()
+      for (let i = 0; i < this.items.length; i += chunkSize) {
+        rows.push(this.items.slice(i, i + chunkSize))
+      }
+      return rows
+    },
+    replaceBlockClass() {
+      const blockClass = document.getElementsByClassName('block')[0]
+      blockClass.className += ` ${this.slice.primary.block_style}`
+    },
```

Comment:
Est-ce qu'on a besoin de mettre ça dans le mounted ? Est-ce qu'on ne peut pas directement mettre la valeur `block_style` dans la class d'un premier div ?

Reply:
La seule façon que je vois de mettre directement la valeur dans la div est de lier dynamiquement les classes, ce qui ferait quelque chose comme : 

`
<template>
    <div :class="{process: isProcess, statistics: isStatistics, features: isFeatures}" />
</template>

<script>
    export default {
    computed: {
        isStatistics() { return this.slice.primary.block_style === 'statistics'},
        isProcess() { return this.slice.primary.block_style === 'process'},
        isFeatures() { return this.slice.primary.block_style === 'features'},
    }
}
</script>
`

Je ne trouve pas cela forcément très maintenable dans le sens où chaque nouveau bloc devrait être inséré à la fois dans une computed property et dans la div d'origine.

Je pense cependant qu'il est possible d'obtenir l'information du CSS dans un hook antérieur au mounted (ex: `beforeMount()`), ce qui pourrait améliorer le temps de rendu du CSS.

Reply:
Effectivement, je me suis compliqué la vie alors qu'il était juste possible de lier la classe de façon dynamique, réglant au passage le problème de latence de chargement du CSS, merci !